src/postgraphql/withPostGraphQLContext.js

```diff
@@ -0,0 +1,45 @@
+import { $$pgClient } from '../postgres/inventory/pgClientFromContext'
+import debugPgClient from './http/debugPgClient'
+import setupPgClientTransaction from './setupPgClientTransaction'
+
+/*
+
+This is intended to be called similar to this:
+
+    const result = await withPostGraphQLContext(options, { pgPool, jwtToken }, async context => (
+      await graphql(
+        schema,
+        query,
+        null,
+        {  ...context },
+        variables,
+        operationName,
+      )
+    ))
+
+*/
+
+export default async function withPostGraphQLContext(options, {pgPool, jwtToken}, functionToRun) {
+  // Connect a new Postgres client and start a transaction.
+  const pgClient = await pgPool.connect()
+
+  // Enhance our Postgres client with debugging stuffs.
+  debugPgClient(pgClient)
+
+  // Begin our transaction and set it up.
+  await pgClient.query('begin')
+  await setupPgClientTransaction(jwtToken, pgClient, options)
```

Comment:
Can we move `setupPgClientTransaction` into this function and then call `withPostGraphQLContext` from [here](https://github.com/calebmer/postgraphql/blob/c43ab8d852b4f83704571287c64662b1a3e6d915/src/postgraphql/http/createPostGraphQLHttpRequestHandler.js#L277-L308). DRY it up a little and allows us to internally dogfood the solution :wink: